src/Algorithm.php

```diff
@@ -18,6 +21,12 @@ public static function create($name)
         case 'hmac-sha256':
             return new HmacAlgorithm('sha256');
             break;
+        case 'rsa-sha1':
```

Comment:
Should we even be supporting sha1, a very old and problematic algo?

Reply:
Yeah fair enough. It might be a good idea to give a warning or even just a comment somewhere that sha1 is insecure and sha256 should be the default choice

Reply:
I'm not a fan of the current instructions for context setup where an array of keys is supplied. We could simplify by instructing a single key be used for the signing context, possibly detecting the key type and determine the algorithm automatically - if it's not RSA or EC, it is a secret, so hmac. If RSA Private Key, RSA. By default we use SHA256. If the algorithm is explicitly specified then use the hash the user has selected.

Either that, or only select the algorithm when creating the context and set SHA256 internally by default, unless the user selects the full algorithm with hash (SHA1, SHA384, SHA512 at user's own risk for security and/or compatibility).

Reply:
For the singing flow, the context should really only have a single key. We can derive the algorithm from that key type (actually required, can't RSA with a secret, useless to HMAC on a private key) and pick the best hash to go with it, so the user doesn't need to specify an algorithm string. Will work on it in a future PR.

Reply:
It seems like the draft specification's authors are in favour of deprecating sha1 support:
https://github.com/w3c-dvcg/http-signatures/issues/20
We can preempt this or wait for a new revision to be published.

Reply:
Yeah lets not add support for a deprecated option. Happy to leave the existing `hmac-sha1` for BC reasons

Reply:
Well, until it's out of the spec I think it should stay in this PR to make sure we're compatible. Trivial to remove it as and when this is formally removed (and since I'm now a contributor there I'll know about it).